torch/_dynamo/variables/user_defined.py

```diff
@@ -2140,9 +2140,10 @@ class MutableMappingVariable(UserDefinedObjectVariable):
     def __init__(self, value, **kwargs):
         super().__init__(value, **kwargs)
         self.generic_dict_vt = variables.ConstDictVariable({})
-        self.mutation_type = (
-            AttributeMutationExisting() if self.source else AttributeMutationNew()
-        )
+        if not kwargs.get("mutation_type"):
+            self.mutation_type = (
+                AttributeMutationExisting() if self.source else AttributeMutationNew()
+            )
```

Comment:
This makes me think `mutation_type` should've been set properly in `super().__init__(...)`, is that not the case?